### ISSUE TITLE
HDF5 libraries fail test suites on Windows, h5py, PyTables

### ISSUE THREAD

**Issue by mrocklin** [ContinuumIO/anaconda-issues#106]
On a windows machine with anaconda installed neither of the `h5py` nor `pytables` libraries pass their test suites. 

This came from @mwiebe 's development system

```
import h5py
h5py.run_tests()

import tables
tables.test()
```
### `h5py`

```
FAILED (failures=34, errors=19, skipped=18, expected failures=1)
```
### `PyTables`

```
======================================================================
ERROR: test05b_modifyColumns (tables.tests.test_nestedtypes.WriteNoReopen)
Modifying two nested columns (modify_columns).
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\tables\tests\common.py", line 154, in newmethod
    return oldmethod(self, *args, **kwargs)
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\tables\tests\test_nestedtypes.py", line 531, in test05b_modifyColumns
    dtype=raCols.dtype)
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\numpy\core\records.py", line 519, in fromarrays
    arrayList = [sb.asarray(x) for x in arrayList]
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\numpy\core\records.py", line 519, in <listcomp>
    arrayList = [sb.asarray(x) for x in arrayList]
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\numpy\core\numeric.py", line 460, in asarray
    return array(a, dtype, copy=False, order=order)
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\tables\table.py", line 3501, in __iter__
    out=buf_slice)
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\tables\table.py", line 1930, in read
    arr = self._read(start, stop, step, field, out)
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\tables\table.py", line 1834, in _read
    bytes_required))
ValueError: output array size invalid, got 8 bytes, need 16 bytes

======================================================================
ERROR: test05b_modifyColumns (tables.tests.test_nestedtypes.WriteReopen)
Modifying two nested columns (modify_columns).
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\tables\tests\common.py", line 154, in newmethod
    return oldmethod(self, *args, **kwargs)
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\tables\tests\test_nestedtypes.py", line 531, in test05b_modifyColumns
    dtype=raCols.dtype)
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\numpy\core\records.py", line 519, in fromarrays
    arrayList = [sb.asarray(x) for x in arrayList]
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\numpy\core\records.py", line 519, in <listcomp>
    arrayList = [sb.asarray(x) for x in arrayList]
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\numpy\core\numeric.py", line 460, in asarray
    return array(a, dtype, copy=False, order=order)
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\tables\table.py", line 3501, in __iter__
    out=buf_slice)
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\tables\table.py", line 1930, in read
    arr = self._read(start, stop, step, field, out)
  File "C:\Users\mwiebe\Anaconda\lib\site-packages\tables\table.py", line 1834, in _read
    bytes_required))
ValueError: output array size invalid, got 8 bytes, need 16 bytes

----------------------------------------------------------------------
Ran 5392 tests in 258.395s

FAILED (errors=2)
```

**Comment by groutr** [comment]
I get failures with pytables only on my windows system.
The test cases are complaining that they can't create hdf5 files for the test.

**Comment by wavetossed** [comment]
I got the same errors on OSX running the PYTables test using the Anaconda distro of Python, i.e.

export PATH="/opt/anaconda/bin:$PATH"
python
Python 2.7.9 |Anaconda 2.2.0 (x86_64)| (default, Dec 15 2014, 10:37:34) 
[GCC 4.2.1 (Apple Inc. build 5577)] on darwin
blah blah ...

import tables
tables.test

This is a Macbook running the latest OSX Yosemite
Given that Macs are more uniform one to another, you might want to borrow a Yosemite Macbook and run this the Anaconda Python distro to track down where it is coming from.

P.S. possible red herring, but I installed Anaconda because I was unable to build PYTables using

pip install tables
on my regular Python which was 2.7 installed from MacPorts. Something to do with globals,h (originating from BSD/GCC) and a config.h Possibly something to do with glib2
